llvm/lib/CodeGen/PreISelIntrinsicLowering.cpp

```diff
@@ -415,9 +416,10 @@ bool PreISelIntrinsicLowering::expandMemIntrinsicUses(Function &F) const {
       GV->setAlignment(Align(16));
       Value *PatternPtr = GV;
       Value *NumBytes = Builder.CreateMul(
-          Builder.getInt64(DL.getTypeSizeInBits(Memset->getValue()->getType()) /
-                           8),
-          Memset->getLength());
+          ConstantInt::get(IntIdxTy,
+                           DL.getTypeSizeInBits(Memset->getValue()->getType()) /
```

Comment:
Not directly related to your change, but shouldn't this be using `getTypeAllocSize` instead of `getTypeSizeInBits / 8`?